Assets/Shopify/UIToolkit/Test/Unit/Editor/TestSingleProductThemeController.cs

```diff
@@ -0,0 +1,109 @@
+﻿namespace Shopify.UIToolkit.Test.Unit {
+    using UnityEngine;
+    using System.Linq;
+    using NUnit.Framework;
+    using NSubstitute;
+    using UnityEngine.TestTools;
+    using Shopify.Unity;
+    using Shopify.Tests;
+    using Shopify.Unity.SDK;
+    using System.Collections;
+    using System.Collections.Generic;
+    using Shopify.Unity.SDK.Editor;
+
+    [TestFixture]
+    public class TestSingleProductThemeController {
+        private SingleProductThemeController _controller;
+        private ISingleProductTheme _theme;
+
+        [SetUp]
+        public void Setup() {
+            _controller = GlobalGameObject.AddComponent<SingleProductThemeController>();
+            _theme = Substitute.For<ISingleProductTheme>();
+
+            _controller.Theme = _theme;
+
+            _controller.ShopDomain = Utils.TestShopDomain;
+            _controller.AccessToken = Utils.TestAccessToken;
+        }
+
+        [Test]
+        public void TestControllerHasProductGID() {
+            _controller.ProductGID = "hello";
+            Assert.AreEqual("hello", _controller.ProductGID);
+        }
+
+        [UnityTest]
+        public IEnumerator TestShowLoadsProductAndTellsTheme() {
+            _controller.ProductGID = "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUyODE0NzU=";
+            _controller.Show();
+
+            var callbackCalled = false;
+
+            _theme.When((x) => {
+                x.OnShouldShowProduct(Arg.Any<Product>(), Arg.Any<ProductVariant[]>());
+            }).Do((x) => {
+                var product = x.Args()[0] as Product;
+                var productVariants = x.Args()[1] as ProductVariant[];
+
+                Assert.AreEqual("Arena Zip Boot", product.title());
+                Assert.AreEqual(10, productVariants.Length);
+
+                callbackCalled = true;
+            });
+
+            while (!callbackCalled) {
+               yield return null;
+            }
+        }
+
+        /* TODO: Fix bug where exception is thrown when productGID is valid but not found on shop
```

Comment:
Oh yea this throws an error in the json deserializer???? I don't wanna forget about this test though so I'll keep it around as a TODO and prioritize this fix

Reply:
I wonder if this is a bug in how we deserialize the JSON when we encounter errors in the GraphQL response. I feel like this should return GraphQL errors saying the product wasn't found.